pkg/metrics/reconciler/reconciler.go

```diff
@@ -0,0 +1,58 @@
+package reconciler
+
+import (
+	"context"
+
+	nebulav1 "github.com/puppetlabs/relay-core/pkg/apis/nebula.puppet.com/v1"
+	"github.com/puppetlabs/relay-core/pkg/metrics/model"
+	"github.com/puppetlabs/relay-core/pkg/operator/obj"
+	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/metric"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type WorkflowRunReconciler struct {
+	client client.Client
+	meter  *metric.Meter
+}
+
+var _ reconcile.Reconciler = &WorkflowRunReconciler{}
+
+func (r *WorkflowRunReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
```

Comment:
Just so everyone's on the same page here, we don't think there's a guarantee that even if you always return no error from a reconciler that you won't end up with the reconciler being invoked twice for the same object in the same state (for example, a trivial change to the labels would cause this to go through the reconcile loop again). However, this is much better than the alternative of putting it into the main reconciler for now, since that one returns transient errors frequently that cause a lot of requeues. So we're putting this into the "good enough" category and we'll see what the numbers end up looking like.